api/src/pcapi/core/users/api.py

```diff
@@ -1114,19 +1114,18 @@ def _filter_user_accounts(accounts: BaseQuery, search_term: str) -> BaseQuery:
     else:
         term_filters.append(models.User.phoneNumber == term_as_phone_number)  # type: ignore[arg-type]
 
+    split_terms = [email_utils.sanitize_email(term) for term in re.split(r"[,;\s]+", search_term) if term]
+
     # numeric (single id or multiple ids)
-    split_terms = re.split(r"[,;\s]+", search_term)
     if all(term.isnumeric() for term in split_terms):
         term_filters.append(models.User.id.in_([int(term) for term in split_terms]))
 
     # email
-    sanitized_term = email_utils.sanitize_email(search_term)
-
-    if email_utils.is_valid_email(sanitized_term):
-        term_filters.append(models.User.email == sanitized_term)
-    elif email_utils.is_valid_email_domain(sanitized_term):
+    if all(email_utils.is_valid_email(term) for term in split_terms):
```

Comment:
Je suis pas sûr que ça arrivera mais si jamais `split_terms == []` on aura `all([]) is True` et donc ça rajoutera potentiellement le filtre `models.User.email.in_([])` ce qui donnera une liste de comptes vide

Reply:
Bien vu, on peut arriver dans ce cas avec une simple virgule dans le champ de recherche : 
```
>>> search_term = ","
>>> split_terms = [term for term in re.split(r"[,;\s]+", search_term) if term]
>>> split_terms
[]
>>> all(term.isnumeric() for term in split_terms)
True
```
Le résultat sera vide, ce que tu prédisais, donc ce sera bien le résultat attendu.
A priori c'est déjà ce qui se passe avant cette PR.
Mais je rajoute un retour plus explicite pour clarifier.